nrfutil/device/program.ts

```diff
@@ -62,33 +62,41 @@ export const isNordicDfuProgrammingOptions = (
     !isMcuBootProgrammingOptions(options) &&
     (options as NordicDfuProgrammingOptions).mcuEndState !== undefined;
 
-export const programmingOptionsToArgs = (options?: ProgrammingOptions) => {
+const programmingOptionsToStrings = (options?: ProgrammingOptions) => {
     if (!options) return [];
 
-    const args: string[] = [];
-
     if (isJLinkProgrammingOptions(options)) {
-        if (options.chipEraseMode)
-            args.push(`chip_erase_mode=${options.chipEraseMode}`);
-        if (options.reset) args.push(`reset=${options.reset}`);
-        if (options.verify) args.push(`verify=${options.verify}`);
-    } else if (isMcuBootProgrammingOptions(options)) {
-        if (options.mcuEndState)
-            args.push(`mcu_end_state=${options.mcuEndState}`);
-        if (options.netCoreUploadDelay)
-            args.push(
+        return [
+            `chip_erase_mode=${options.chipEraseMode}`,
```

Comment:
Why does this not have the boolean evaluator?

Reply:
According to [the type predicate `isJLinkProgrammingOptions`](https://github.com/NordicSemiconductor/pc-nrfconnect-shared/blob/main/nrfutil/device/program.ts#L45-L48) `chip_erase_mode` must always be defined if the `options` are of type `JLinkProgrammingOptions`. I also changed the field in the type to be mandatory.